ExcelFile.parse() is missing

**Describe the bug**
With pandas-stubs 3.0.3.260530, valid calls to ExcelFile.parse() raise errors with mypy.

**To Reproduce**
*1. Provide a minimal runnable `pandas` example that is not properly checked by the stubs.*

This requires a file.xlsx in Excel format with a single sheet named "sheet", for instance like [file.xlsx](https://github.com/user-attachments/files/29205892/file.xlsx).
```py
import pandas

ef = pandas.ExcelFile("file.xlsx")
print(ef.parse("sheet"))  # Call the method
print(ef.parse.__doc__)   # Show its docstring
```
This gives output like:
```
$ python bug.py
Empty DataFrame
Columns: []
Index: []

Parse specified sheet(s) into a DataFrame.

Equivalent to read_excel(ExcelFile, ...)  See the read_excel
docstring for more info on accepted parameters.
...
```

*2. Indicate which type checker you are using (`mypy` or  `pyright`).*
```
$ mypy -V
mypy 2.1.0 (compiled: yes)
```

*3. Show the error message received from that type checker while checking your example.*

The minimal example passes type checking with pandas-stubs 3.0.0.260204.
```
$ uv pip install pandas-stubs
…
 + pandas-stubs==3.0.3.260530
~$ mypy bug.py
bug.py:4: error: "ExcelFile" has no attribute "parse"  [attr-defined]
bug.py:5: error: "ExcelFile" has no attribute "parse"  [attr-defined]
Found 2 errors in 1 file (checked 1 source file)
~$ uv pip install "pandas-stubs < 3.0.3.260530"
…
 - pandas-stubs==3.0.3.260530
 + pandas-stubs==3.0.0.260204
~$ mypy bug.py
Success: no issues found in 1 source file
```

**Please complete the following information:**
- OS, OS Version, python version: 
  ```
  $ lsb_release -d && python -V
  Description:	Ubuntu 26.04 LTS
  Python 3.14.0
  ```
- version of type checker: 2.1.0, shown above
- version of installed `pandas-stubs`: 3.0.3.260530, shown above.

**Additional context**
Looking at the history of pandas-stubs/io/excel/_base.pyi, I see that hints for this method existed but were removed by #1709. That PR's description includes:

> Organic programming, removing the deprecated parse method.

…but otherwise seems incomplete. There is a comment "/pandas_nightly", and no references or links. To me as a user this is cryptic, but maybe it implies that ExcelFile.parse() *will* be deprecated in a future version of pandas.

The release notes for [pandas 3.0.3](https://pandas.pydata.org/docs/whatsnew/v3.0.3.html) and [pandas 3.0.0](https://pandas.pydata.org/docs/whatsnew/v3.0.0.html#deprecations) don't mention such a deprecation, and there is no deprecation warning given when running the example above. According to the [pandas version policy](https://pandas.pydata.org/docs/development/policies.html#version-policy), this means the method is not deprecated in any existing release, and the earliest release in which it could be deprecated would be 3.1.0.

I've read [this part of the pandas-stubs README](https://github.com/pandas-dev/pandas-stubs#compatibility-with-pandas), which says:

> Type checkers will flag deprecated or removed APIs based on the pandas version the stubs were tested against.

I would interpret that since ExcelFile.parse() is not deprecated or removed in pandas 3.0.3, pandas-stubs 3.0.3.x should not error.

--- a/pandas-stubs/io/excel/_base.pyi
+++ b/pandas-stubs/io/excel/_base.pyi
@@ -305,6 +305,62 @@ class ExcelFile:
         engine_kwargs: dict[str, Any] | None = ...,
     ) -> None: ...
     def __fspath__(self) -> str: ...
+    @overload
+    def parse(
+        self,
+        sheet_name: list[int | str] | None,
+        header: int | Sequence[int] | None = ...,
+        names: ListLikeHashable | None = ...,
+        index_col: int | Sequence[int] | None = ...,
+        usecols: str | UsecolsArgType = ...,
+        converters: dict[int | str, Callable[[Any], Any]] | None = ...,
+        true_values: Iterable[Hashable] | None = ...,
+        false_values: Iterable[Hashable] | None = ...,
+        skiprows: int | Sequence[int] | Callable[[object], bool] | None = ...,
+        nrows: int | None = ...,
+        na_values: Sequence[str] | dict[str | int, Sequence[str]] = ...,
+        parse_dates: (
+            bool
+            | Sequence[int]
+            | Sequence[Sequence[str] | Sequence[int]]
+            | dict[str, Sequence[int] | list[str]]
+        ) = ...,
+        date_parser: Callable[..., Any] | None = ...,
+        thousands: str | None = ...,
+        comment: str | None = ...,
+        skipfooter: int = ...,
+        keep_default_na: bool = ...,
+        na_filter: bool = ...,
+        **kwds: Any,
+    ) -> dict[int | str, DataFrame]: ...
+    @overload
+    def parse(
+        self,
+        sheet_name: int | str,
+        header: int | Sequence[int] | None = ...,
+        names: ListLikeHashable | None = ...,
+        index_col: int | Sequence[int] | None = ...,
+        usecols: str | UsecolsArgType = ...,
+        converters: dict[int | str, Callable[[Any], Any]] | None = ...,
+        true_values: Iterable[Hashable] | None = ...,
+        false_values: Iterable[Hashable] | None = ...,
+        skiprows: int | Sequence[int] | Callable[[object], bool] | None = ...,
+        nrows: int | None = ...,
+        na_values: Sequence[str] | dict[str | int, Sequence[str]] = ...,
+        parse_dates: (
+            bool
+            | Sequence[int]
+            | Sequence[Sequence[str] | Sequence[int]]
+            | dict[str, Sequence[int] | list[str]]
+        ) = ...,
+        date_parser: Callable[..., Any] | None = ...,
+        thousands: str | None = ...,
+        comment: str | None = ...,
+        skipfooter: int = ...,
+        keep_default_na: bool = ...,
+        na_filter: bool = ...,
+        **kwds: Any,
+    ) -> DataFrame: ...
     @property
     def book(self) -> Workbook | Book | OpenDocument | PyXlsbWorkbook: ...
     @property

--- a/tests/test_io.py
+++ b/tests/test_io.py
@@ -1118,9 +1118,12 @@ def test_excel_writer(tmp_path: Path) -> None:
     check(assert_type(ef, pd.ExcelFile), pd.ExcelFile)
     check(assert_type(read_excel(ef, sheet_name="A"), DataFrame), DataFrame)
     check(assert_type(read_excel(ef), DataFrame), DataFrame)
+    check(assert_type(ef.parse(sheet_name=0), DataFrame), DataFrame)
+    check(
+        assert_type(ef.parse(sheet_name=[0]), dict[str | int, DataFrame]),
+        dict,
+    )
     check(assert_type(ef.close(), None), type(None))
-    if TYPE_CHECKING_INVALID_USAGE:
-        _ = ef.parse  # type: ignore[attr-defined] # pyright: ignore[reportAttributeAccessIssue,reportUnknownMemberType,reportUnknownVariableType] # pyrefly: ignore[missing-attribute]
 
 
 def test_excel_writer_io() -> None:

--- a/tests/test_io.py
+++ b/tests/test_io.py
@@ -1119,11 +1119,25 @@ def test_excel_writer(tmp_path: Path) -> None:
     check(assert_type(ef, pd.ExcelFile), pd.ExcelFile)
     check(assert_type(read_excel(ef, sheet_name="A"), DataFrame), DataFrame)
     check(assert_type(read_excel(ef), DataFrame), DataFrame)
-    check(assert_type(ef.parse(sheet_name=0), DataFrame), DataFrame)
-    check(
-        assert_type(ef.parse(sheet_name=[0]), dict[str | int, DataFrame]),
-        dict,
-    )
+
+    with pytest_warns_bounded(
+        errors.Pandas4Warning,
+        match="ExcelFile.parse is deprecated",
+        lower="3.0.99",
+        upper="3.99",
+    ):
+        check(assert_type(ef.parse(sheet_name=0), DataFrame), DataFrame)
+
+    with pytest_warns_bounded(
+        errors.Pandas4Warning,
+        match="ExcelFile.parse is deprecated",
+        lower="3.0.99",
+        upper="3.99",
+    ):
+        check(
+            assert_type(ef.parse(sheet_name=[0]), dict[str | int, DataFrame]),
+            dict,
+        )
     check(assert_type(ef.close(), None), type(None))
 
 

--- a/tests/test_io.py
+++ b/tests/test_io.py
@@ -288,6 +288,7 @@ def test_clipboard_iterator() -> None:
 
 
 def test_sas_bdat() -> None:
+    # there is a bug with pandas 3.1.0.dev0+1139.gb431b4f927 that SegFault the code
     path = Path(CWD, "data", "airline.sas7bdat")
     check(assert_type(read_sas(path), DataFrame), DataFrame)
     with check(

